config/tests/samples/create/samples.go

```diff
@@ -392,7 +392,34 @@ func ListMatchingSamples(t *testing.T, regex *regexp.Regexp) []SampleKey {
 				sampleKey := samples[filepath.Dir(path)]
 				sampleKey.Name = sampleName
 				sampleKey.SourceDir = filepath.Dir(path)
-				sampleKey.files = append(sampleKey.files, path)
+
+				// Heuristic to find the "main" resource for the sample, which we define as the resource
+				// whose group matches the service name in the path.
+				// We will prepend this to the list of files so that it is created first.
+				shouldPrepend := false
+				b := test.MustReadFile(t, path)
+				yamls := testyaml.SplitYAML(t, b)
+				if len(yamls) > 0 {
+					u := &unstructured.Unstructured{}
+					if err := yaml.Unmarshal(yamls[0], u); err == nil {
+						gvk := u.GroupVersionKind()
+						apiGroup := gvk.Group
+
+						if strings.HasSuffix(apiGroup, ".cnrm.cloud.google.com") {
+							service := strings.TrimSuffix(apiGroup, ".cnrm.cloud.google.com")
+							if strings.Contains(path, "/"+service+"/") {
```

Comment:
I don't think currently we have "service" in the file path? If we had, it might be much easier to group samples.

Reply:
we do, at least required in README.sample "The sub-folder name should be all lowercase
using dashes instead of spaces, and should be a grammatical description of the
sample type, with the literal resource name included. " (resource name contains service name)